tests/unit_tests/test_server_client.py

```diff
@@ -749,6 +750,132 @@ class Foo(BaseModel):
     assert Model.__name__ == "Bar"
 
 
+@pytest.mark.asyncio
+async def test_input_config_output_schemas(event_loop: AbstractEventLoop) -> None:
+    """Test schemas returned for different configurations."""
+    # TODO(Fix me): need to fix handling of global state -- we get problems
+    # gives inconsistent results when running multiple tests / results
+    # depending on ordering
+    server._SEEN_NAMES = set()
```

Comment:
fyi @nfcampos  -- I'll need to fix this up. keeping for now since easy and it adds a bit of testing coverage